common-app-lib/src/cmr/common_app/services/search/elastic_search_index.clj

```diff
@@ -48,39 +49,101 @@
   [concept-type fields]
   (map #(q2e/query-field->elastic-field (keyword %) concept-type) fields))
 
+(defn- query->execution-params
+  "Returns the Elasticsearch execution parameters extracted from the query. These are the
+  actual ES parameters as epxected by the elastisch library. The :scroll-id parameter is special
+  and is stripped out before calling elastisch to determine whether a normal search call or a
+  scroll call should be made."
+  [query]
+  (let [{:keys [page-size offset concept-type aggregations highlights scroll scroll-id]} query
```

Comment:
I'm sure you guys have thought of this, but I'll bring it up so you can reassure me: this is a prototypical entry point for injection attacks ... there's another point in the code below where I'm guessing there might be protections for this? I'll comment there, too.